util/signing/interface/src/cryptography/ed25519.rs

```diff
@@ -0,0 +1,36 @@
+use crate::cryptography::Curve;
+use crate::{Verify, VerifyError};
+use anyhow::Context;
+use ring_compat::signature::{ed25519, Verifier};
+
+/// The Ed25519 curve.
+#[derive(Debug, Clone)]
+pub struct Ed25519;
+
+fixed_size!(pub struct PublicKey([u8; 32]));
```

Comment:
How does `fixed_sized` help here?

Reply:
It's a macro I defined in `cryptography/mod.rs`, to have one place to write the boilerplate for these fixed-sized types. The types are specific to their elliptic curves to provide type safety against mixing different types of keys or signatures, so they have each their own definition, with no type aliasing or const generics to reuse.